exporter_test.go

```diff
@@ -74,6 +75,44 @@ func TestInvalidUtf8InDatadogTagValue(t *testing.T) {
 	}
 }
 
+type MockHistogram struct {
+	prometheus.Metric
+	prometheus.Collector
+	value float64
+}
+
+func (h *MockHistogram) Observe(n float64) {
+	h.value = n
+
+}
+
+func TestHistogramUnits(t *testing.T) {
+	events := make(chan Events, 1)
+	c := Events{
+		&TimerEvent{
+			metricName: "foo",
+			value:      300,
+		},
+	}
+	events <- c
+	ex := NewExporter(&metricMapper{}, true)
+	ex.mapper.Defaults.TimerType = timerTypeHistogram
+
+	// Close channel to signify we are done with the listener after a short period.
+	go func() {
+		time.Sleep(time.Millisecond * 100)
+		close(events)
+	}()
+	mock := &MockHistogram{}
+	ex.Histograms.Elements[7787632782521330630] = mock
```

Comment:
That is the hash of the metric, it is how all the metrics are stored internally. I suppose I could call out the hash function to generate it.